### PR TITLE
chore: suppress sandbox bin logs

### PR DESCRIPTION
**Pull request by ChaoticTempest**
This suppresses all sandbox logs and allows the user to control it via an environment variable

### CODE REVIEW

File: workspaces/src/network/server.rs [hunk 1]
```diff
@@ -67,3 +70,17 @@ impl Drop for SandboxServer {
...
 }
+
+/// Turn off neard-sandbox logs by default. Users can turn them back on with
+/// NEAR_ENABLE_SANDBOX_LOG=on and specify further paramters with the custom
+/// NEAR_SANDBOX_LOG for higher levels of specificity. NEAR_SANDBOX_LOG args
+/// will be forward into RUST_LOG environment variable as to not conflict
+/// with similar named log targets.
+fn supress_sandbox_logs_if_required() {
+    match std::env::var("NEAR_ENABLE_SANDBOX_LOG") {
+        Ok(val) if val == "on" => {}
```
Comment (austinabell) [inline workspaces/src/network/server.rs:81]: why not just require the env variable to be set? is the `on` pattern documented somewhere?

**Review by austinabell — APPROVED** [review]
didn't functionally test but lgtm

File: workspaces/src/network/server.rs (hunk 1 above)
Comment (itegulov) [reply workspaces/src/network/server.rs:81]: I agree, the usual convention is either set/unset or 1/0

**Review by itegulov — APPROVED** [review]
Tested, works fine

File: workspaces/src/network/server.rs (hunk 1 above)
Comment (ChaoticTempest) [reply workspaces/src/network/server.rs:81]: Ahh, will switch to just set/unset. Way too use to `cmake` flags so a force of habit

File: workspaces/src/network/server.rs [hunk 2]
```diff
@@ -67,3 +70,16 @@ impl Drop for SandboxServer {
             .unwrap();
     }
 }
+
+/// Turn off neard-sandbox logs by default. Users can turn them back on with
+/// NEAR_ENABLE_SANDBOX_LOG=on and specify further paramters with the custom
```
Comment (miraclx) [inline workspaces/src/network/server.rs:75]:
```suggestion
/// NEAR_ENABLE_SANDBOX_LOG=1 and specify further paramters with the custom
```

File: workspaces/src/network/server.rs [hunk 3]
```diff
@@ -67,3 +70,16 @@ impl Drop for SandboxServer {
...
+/// NEAR_SANDBOX_LOG for higher levels of specificity. NEAR_SANDBOX_LOG args
+/// will be forward into RUST_LOG environment variable as to not conflict
+/// with similar named log targets.
+fn supress_sandbox_logs_if_required() {
+    if let Err(_) = std::env::var("NEAR_ENABLE_SANDBOX_LOG") {
+        // non-exhaustive list of targets to supress, since choosing a default LogLevel
+        // does nothing in this case, since nearcore seems to be overriding it somehow:
+        std::env::set_var("NEAR_SANDBOX_LOG", "near=error,stats=error,network=error");
+    };
```
Comment (miraclx) [inline workspaces/src/network/server.rs:84]: With this, `NEAR_ENABLE_SANDBOX_LOG=0` would still activate sandbox logging.

**Pull request merged**